plugin/log/setup.go

```diff
@@ -74,28 +76,36 @@ func logParse(c *caddy.Controller) ([]Rule, error) {
 			rules = append(rules, Rule{
 				NameScope: dns.Fqdn(args[0]),
 				Format:    format,
+				Class:     make(map[response.Class]bool),
 			})
 		}
 
 		// Class refinements in an extra block.
 		for c.NextBlock() {
 			switch c.Val() {
-			// class followed by all, denial, error or success.
+			// class followed by combinations of all, denial, error and success.
 			case "class":
+				if len(rules[len(rules)-1].Class) != 0 {
```

Comment:
what does this actually do? And why doesn't it return an error?

Reply:
Potentially we could have several "class" lines - here we handle only first record (previously we applied the latest)